src/client/datascience/context/telemetry.ts

```diff
@@ -75,13 +75,69 @@ export function getErrorClassification(error: Error) {
         return 'jupyterconnection';
     } else if (isErrorType(error, JupyterInstallError)) {
         return 'jupyterinstall';
+    } else if (isErrorType(error, PythonKernelDiedError)) {
+        return getReasonForKernelToDie(error);
```

Comment:
Beginning to see a lot of benefit in creating specific error classes for each error.
Like in .NET & other langauges. Feel we should do this more often instead of just throwing  a generic error `new Error('xyz')`.